stdlib/LinearAlgebra/test/uniformscaling.jl

```diff
@@ -330,4 +330,12 @@ end
     @test I(3) == [1 0 0; 0 1 0; 0 0 1]
 end
 
+@testset "generalized dot" begin
+    x = rand(-10:10, 3)
+    y = rand(-10:10, 3)
+    λ = rand(-10:10)
+    J = UniformScaling(λ)
+    @test dot(x, J, y) == λ*dot(x, y)
+end
```

Comment:
Probably should also have a test using the `Quaternion` type from testhelpers, similar to the tests in `LinearAlgebra/test/generic.jl`, to make sure we cover the non-commutative case.

Reply:
I agree, but I’m hesitant to write tests blindly via the web interface. 🤣 I’ll do that later when I have my usual working environment around me.

Reply:
Done.